src/NewsService.Business/Commands/Channels/CreateChannelCommand.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using LT.DigitalOffice.Kernel.BrokerSupport.AccessValidatorEngine.Interfaces;
+using LT.DigitalOffice.Kernel.Constants;
+using LT.DigitalOffice.Kernel.Enums;
+using LT.DigitalOffice.Kernel.FluentValidationExtensions;
+using LT.DigitalOffice.Kernel.Helpers.Interfaces;
+using LT.DigitalOffice.Kernel.Responses;
+using LT.DigitalOffice.NewsService.Business.Commands.Channels.Interfaces;
+using LT.DigitalOffice.NewsService.Data.Interfaces;
+using LT.DigitalOffice.NewsService.Mappers.Db.Interfaces;
+using LT.DigitalOffice.NewsService.Models.Dto.Requests.Channel;
+using LT.DigitalOffice.NewsService.Validation.Channel.Interfaces;
+using Microsoft.AspNetCore.Http;
+
+namespace LT.DigitalOffice.NewsService.Business.Commands.Channels
+{
+  public class CreateChannelCommand : ICreateChannelCommand
+  {
+    private readonly IAccessValidator _accessValidator;
+    private readonly IResponseCreator _responseCreator;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IChannelRepository _channelRepository;
+    private readonly IDbChannelMapper _mapper;
+    private readonly ICreateChannelRequestValidator _validator;
+
+    public CreateChannelCommand(
+      IAccessValidator accessValidator,
+      IResponseCreator responseCreator,
+      IHttpContextAccessor httpContextAccessor,
+      IChannelRepository channelRepository,
+      IDbChannelMapper mapper,
+      ICreateChannelRequestValidator validator
+    )
+    {
+      _accessValidator = accessValidator;
+      _responseCreator = responseCreator;
+      _httpContextAccessor = httpContextAccessor;
+      _channelRepository = channelRepository;
+      _mapper = mapper;
+      _validator = validator;
+    }
+
+    public async Task<OperationResultResponse<Guid>> ExecuteAsync(CreateChannelRequest request)
+    {
+      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveNews))
+      {
+        return _responseCreator.CreateFailureResponse<Guid>(HttpStatusCode.Forbidden);
+      }
+
+      if (!_validator.ValidateCustom(request, out List<string> errors))
+      {
+        return _responseCreator.CreateFailureResponse<Guid>(HttpStatusCode.BadRequest, errors);
+      }
+
+      OperationResultResponse<Guid> response = new();
+
+      _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Created;
+
+      response.Body = await _channelRepository.CreateAsync(await _mapper.MapAsync(request));
+
+      response.Status = response.Errors.Any()
```

Comment:
how do you get errors if you don't add them anywhere?

Reply:
fixed